TinkeringGraphics

```diff
@@ -0,0 +1,111 @@
+#To start the prog type:   main()
+
+#Turns the image black and white
+def blackAndWhite(image):
+   for pixel in image.getPixels():
+      red = pixel.getRed()
+      green = pixel.getGreen()
+      blue = pixel.getBlue()
+      average = (red + green + blue) / 3
+      if average < 128:
+         red = green = blue = 0
+      else:
+         red = green = blue = 255
+      pixel.setRed(red)
+      pixel.setGreen(green)
+      pixel.setBlue(blue)
+#add a grid to the image
+def grid(pic):
+  width = getWidth(pic)
+  height = getHeight(pic)
+  size = 10
+  for x in range(0, width, size):
+    addLine(pic, x+size, 0, x+size, height)
+  for y in range(0, height, size):
+    addLine(pic, 0, y+size, width, y+size)
+
+def colorFilter(image, rgbTriple):
+   def baseValue(value, offset):
```

Comment:
Avoid defining functions inside of other functions, unless there is a compelling reason to restrict access.